streams/shared/src/main/scala/zio/stream/ZSink.scala

```diff
@@ -824,6 +824,9 @@ object ZSink extends ZSinkPlatformSpecificConstructors {
       } yield push
     }
 
+  def managed[R, E, I, A, L, Z](resource: ZManaged[R, E, A])(fn: A => ZSink[R, E, I, L, Z]): ZSink[R, E, I, L, Z] =
```

Comment:
this signature is much simpler 👍

Reply:
yeah, I started with the original signature from the ticket, then tried to use it in `fromFile` and it immediately folded. Nested resources are too useful.